jaseci_kit/jaseci_kit/modules/entity_extraction_type_2/entity_extraction.py

```diff
@@ -64,38 +137,24 @@ def train(
         }
         raise HTTPException(status_code=400, detail=st)
 
-    data = pd.DataFrame(columns=["text", "annotation"])
-    if train_data:
-        for t_data in train_data:
-            tag = []
-            for ent in t_data["entities"]:
-                if ent["entity_value"] and ent["entity_type"]:
-                    tag.append(
-                        (
-                            ent["entity_value"],
-                            ent["entity_type"],
-                            ent["start_index"],
-                            ent["end_index"],
-                        )
-                    )
-                else:
-                    raise HTTPException(
-                        status_code=404, detail=str("Entity Data missing in request")
-                    )
-            data = data.append(
-                {"text": t_data["context"], "annotation": tag}, ignore_index=True
-            )
-        # creating training data
-        try:
-            completed = create_data(data)
-        except Exception as e:
-            completed = create_data1(data)
-            print(f"Exception  : {e}")
+    # data = pd.DataFrame(columns=["text", "annotation"])
+    if len(dev_data) != 0:
+        create_train_data(dev_data, "dev")
+
+    if len(test_data) != 0:
+        create_train_data(test_data, "test")
+
+    if len(train_data) != 0:
+        completed = create_train_data(train_data, "train")
```

Comment:
We'll always need the train file for training, so this should raise exception if null

Reply:
yes its already raised exception if not get train data